### ISSUE TITLE
Authentication failed with IBEAM_MAX_FAILED_AUTH=5

### ISSUE THREAD

**Issue by cpawelzik** [Voyz/ibeam#61]
First I'd like to thank you for this great project!

I'm testing the current `v0.4.0` branch with a paper trading account. For some reason, the authentication failed this morning and then ibeam reached the maximum number of authentication attempts. 

Container logs:

> 2022-02-13 08:36:41,388|I| Gateway running and authenticated
2022-02-13 08:37:41,369|I| Gateway running and authenticated
2022-02-13 08:38:41,383|I| Gateway running and authenticated
2022-02-13 08:39:41,402|I| Gateway session found but not authenticated, authenticating...
2022-02-13 08:39:50,688|E| Error displayed by the login webpage: Invalid username password combination
2022-02-13 08:39:57,880|E| Error displayed by the login webpage: Invalid username password combination
2022-02-13 08:40:04,969|E| Error displayed by the login webpage: Invalid username password combination
2022-02-13 08:40:12,057|E| Error displayed by the login webpage: Invalid username password combination
2022-02-13 08:40:19,161|E| Error displayed by the login webpage: Invalid username password combination
2022-02-13 08:40:19,197|E| ######## ATTENTION! ######## Maximum number of failed authentication attempts (IBEAM_MAX_FAILED_AUTH=5) reached. IBeam will shut down to prevent an account lock-out. It is recommended you attempt to authenticate manually in order to reset the counter. Read the execution logs and report issues at https://github.com/Voyz/ibeam/issues
2022-02-13 08:40:19,280|I| Authentication process failed
2022-02-13 08:40:19,280|W| Shutting IBeam maintenance down due to exceeded number of failed attempts.

I restarted the container and now it works again. 

Because this happened on a sunday morning, I believe that it was due to IB restarting their systems, etc. 

Does ibeam also restart the gateway process if such an error happens? If it reaches `IBEAM_MAX_FAILED_AUTH`, would it make sense to wait for 30 minutes or so, restart the gateway, reset the counter and then try to authenticate again?

**Comment by Voyz** [comment]
Hello @cpawelzik thanks for the kind words and welcome to IBeam 👋 Also thanks for outlining your case in detail, let me address:

Assuming that you haven't changed your credentials between yesterday morning and later in the afternoon, I'd say that the problem lays on IBKR side and I'd recommend you open a support ticket to make them aware of your issue. Indeed it could have been them restarting their servers, which - in my opinion - is something that nevertheless should ideally not lead to potentially account-locking results.

> Does ibeam also restart the gateway process if such an error happens

Once IBeam detects consecutive failed attempts exceeding `IBEAM_MAX_FAILED_AUTH` it just shuts down and doesn't restart. 

> would it make sense to wait for 30 minutes or so, restart the gateway, reset the counter and then try to authenticate again?

Note, that the counter really lays on IBKR server side, not on IBeam side - IBeam just tries to estimate it. 

If we did what you outline here, and further attempts would exhibit same type of problems causing failures, it could lead to the account being locked out. IBeam specifically attempts to avoid that by shutting down and displaying the following message:

```
It is recommended you attempt to authenticate manually in order to reset the counter.
```

Therefore, the responsibility to restore the correct functionality lays back on human side. The steps would be:

1. Start the Gateway manually on your development machine
2. Authenticate successfully using the web browser - this resets the counter.
3. Restart IBeam manually.

This way you can ensure that the account lock-out has been evaded. 

> I restarted the container and now it works again.

While this is similar to resetting the lock-out counter manually, this is a risky method that I wouldn't recommend. If the `Invalid username password combination` continued to happen, IBeam would reattempt authentication promptly, potentially leading to the lock-out within about one minute.

----

In conclusion, I'd recommend against the suggestion you propose - I hope you can see why. Let me know if you'd have any comments or suggestions here 👍 Voy

**Comment by cpawelzik** [comment]
Hello @Voyz,

Thanks for your thorough answer. 

I agree that it would be too risky to reset the counter and an account-lock could be the consequence. Currently, the client portal gateway seems to be quite unstable. During my tests, it canceled the brokerage session when I sent an invalid request. Let's hope that IB improves it in future releases. 

Best, 
Chris

**Comment by Voyz** [comment]
Thanks @cpawelzik - indeed hoping IBKR will improve things here and there. Best of luck!

**Comment by elaamrani** [comment]
@Voyz @cpawelzik I am having the same issue with gateway running and authenticated then Invalid username password combination. The problem is indeed from IBKR, but I don't know what exactly.

**Comment by Voyz** [comment]
@elaamrani thanks for sharing your issue too. I'd recommend getting in touch with IBKR support to make them more aware of this problem.
